### ISSUE TITLE
Install `nelson-minimalist-core` in local environment

### ISSUE THREAD

**Issue by Hind-M** [Nelson-numerical-software/nelson-minimalist-core#8]
Hello,
Thanks for this great work! I want to use `nelson-minimalist-core` as a package, and tried to install it but it seems that there are some missing files (some of them are available in the original `nelson` (`gpl-3.0.md`  `CHANGELOG-0.{1, 2, 3, 4, 5}.x.md` `modules/{module_name}/etc` ) and some aren't (`modules/modules.m`)). I wanted to check if I missed something or if this is a mistake that needs fixing. I tried though to comment the corresponding lines in `CMakeLists.txt ` files and managed to get an install but I don' find any files to include in the `ROOT_OUTPUT` apart from libs and executables...
Thanks!

**Comment by Nelson-numerical-software** [comment]
Good point !!!
nelson-minimalist-core was not checked about install.
I will finish release of 0.6.12 (tomorrow) and I will update CmakeList.txt and add in Github CI.
"modules/modules.m" is a generated file in main repo. in nelson-minimalist-core there is no modules.m currently.

on local build, you can call 'bin/linux/nelson' script

**Comment by Nelson-numerical-software** [comment]
see https://github.com/Nelson-numerical-software/nelson-minimalist-core/tree/feat-%238-install-local
![image](https://user-images.githubusercontent.com/23081266/209695676-5f544952-4228-480c-b2cf-4fe6f30e49fc.png)

**Comment by Hind-M** [comment]
Great, thanks!

> on local build, you can call 'bin/linux/nelson' script

Yes, but I meant including nelson header files and using functions inside my cpp program directly? Is that possible?

**Comment by Nelson-numerical-software** [comment]
Currently, I did not define minimal headers list required to call nelson as a pure C/C++ library.
I do not say that it is not possible, only that I did not the job for it ...

In nelson, the way that I expose nelson's engine is to use mex as engine but here we do not have mex.
https://nelson-9.gitbook.io/nelson/en/mex
https://github.com/Nelson-numerical-software/nelson/blob/master/modules/mex/examples/mex_engine_demo_1.c

basically, I could expose:
https://github.com/Nelson-numerical-software/nelson-minimalist-core/tree/master/modules/engine/src/include
and
https://github.com/Nelson-numerical-software/nelson-minimalist-core/tree/master/modules/types/src/include

Can you help me by telling me what you want to do?
and I'll try to tell you how to do in nelson :)

I will try to do a basic example with nelson to call as engine without mex.

**Comment by Hind-M** [comment]
Yes, so I was thinking more of exposing
https://github.com/Nelson-numerical-software/nelson-minimalist-core/tree/master/modules/interpreter/src/include
since I want to use the "interpreter" and evaluate the given commands, check the status/output of evaluation etc, so I guess I should have access to `Evaluator`?

**Comment by Nelson-numerical-software** [comment]
Done to "expose" headers of interpreter, engine, and types for some experimental developpments.
But Please notice that there is no reason to use evaluator directly.
Evaluator is and will be private and can change in few months.
We will define a clean API
with start, stop engine, create a variable, execute a command, ...

Can you define your first need, I will use it as prerequisite for a clean example ;)

**Comment by Nelson-numerical-software** [comment]
@Hind-M , Please confirm that PR is currently enough before I merge it ? :)

**Comment by Nelson-numerical-software** [comment]
I do not know what you want to do, but if you want to redirect input/output.
The better way is to create a new Input/Output interface and uses it.
see https://github.com/Nelson-numerical-software/nelson-minimalist-core/blob/master/modules/stream_manager/src/include/Interface.hpp
here an example of Input/Output using socketio: https://github.com/Nelson-numerical-software/nelson/blob/master/modules/sio_client/src/include/SioClientInterface.hpp

**Comment by Hind-M** [comment]
So what I really want to do at the end is to implement a `xeus-nelson`  (a jupyter kernel for nelson) like https://github.com/jupyter-xeus/xeus-cling for `C++` or https://github.com/jupyter-xeus/xeus-python for `python`.
So to implement https://github.com/jupyter-xeus/xeus-cling/blob/main/src/xinterpreter.cpp#L65 for example, we use the cling interpreter to evaluate the given commands:
https://github.com/jupyter-xeus/xeus-cling/blob/main/src/xinterpreter.cpp#L112-L116
